File: two_sum/java/two_sum.java
public class two_sum {
        public static void twoSumOptimized(int[] arr, int target) {
            for (int i = 0; i < arr.length; i++) {
                int iteratorA = arr[i];
                if (target > iteratorA) {
                    for (int j = 0; j < arr.length; j++) {
                        int iteratorB = arr[j];
                        if (iteratorB + iteratorA == target) {
                            System.out.println(i + " " + j);
                        }
                    }
                }
            }
        }
        
        public static void main(String[] args) {
            int[] arr = {1, 44, 32, 11};
            int target = 12;
            twoSumOptimized(arr, target);
        }
    
}

File: two_sum/twoSum.py
def two_sum(nums, target):
    lookup = {}
    for i, num in enumerate(nums):
        complement = target - num
        if complement in lookup:
            return [lookup[complement], i]
        lookup[num] = i
    return []


target = int(input())
arr  = list(map(int, input().split()))
print(' '.join(map(str, two_sum(arr, target))))



